File: src/main/java/galaxyoyo/unknown/api/editor/Case.java
package galaxyoyo.unknown.api.editor;

import galaxyoyo.unknown.api.editor.sprites.Sprite;

public class Case
{
	private int x;
	private int y;
	private Sprite couche1;
	private Sprite couche2;
	private Sprite couche3;
	private Collision collision;
	
	public int getPosX()
	{
		return x;
	}
	
	public int getPosY()
	{
		return y;
	}
	
	public Sprite getCoucheOne()
	{
		return couche1;
	}
	
	public Sprite getCoucheTwo()
	{
		return couche2;
	}
	
	public Sprite getCoucheThree()
	{
		return couche3;
	}
	
	public Collision getCollision()
	{
		return collision;
	}
	
	public static Case create(int posX, int posY, Sprite couche1,  Sprite couche2, Sprite couche3, Collision collision)
	{
		Case c = new Case();
		c.x = posX;
		c.y = posY;
		c.couche1 = couche1;
		c.couche2 = couche2;
		c.couche3 = couche3;
		c.collision = collision;
		return c;
	}
}

File: src/main/scala/fr/ynerant/leveleditor/api/editor/Case.scala
package fr.ynerant.leveleditor.api.editor

import fr.ynerant.leveleditor.api.editor.Collision.Collision
import fr.ynerant.leveleditor.api.editor.sprites.Sprite


object Case {
	def create(posX: Int, posY: Int, couche1: Sprite, couche2: Sprite, couche3: Sprite, collision: Collision): Case = {
		val c = new Case
		c.x = posX
		c.y = posY
		c.couche1 = couche1
		c.couche2 = couche2
		c.couche3 = couche3
		c.collision = collision
		c
	}
}

class Case {
	private var x = 0
	private var y = 0
	private var couche1 = null: Sprite
	private var couche2 = null: Sprite
	private var couche3 = null: Sprite
	private var collision = null: Collision

	def getPosX: Int = x

	def getPosY: Int = y

	def getCoucheOne: Sprite = couche1

	def getCoucheTwo: Sprite = couche2

	def getCoucheThree: Sprite = couche3

	def getCollision: Collision = collision

	override def toString: String = "{Case x=" + x + " y=" + y + " couche1=" + couche1 + " couche2=" + couche2 + " couche3=" + couche3 + " collision=" + collision.toString.toUpperCase + "}\n"
}
